Topic recognition: 47444243 ViT for ADNI dataset

STommy64 opened the pull request:
Visual Transformer to classify MRI scans in the ADNI dataset as "Alzheimer's disease" or "normal cognition".
Contains the following files:

- dataset.py: Used to generate the dataloaders for training, validating, and testing given a file path, assumes that a 'train' and 'test' folder exists
- modules.py: Contains the ViT model and all submodules
- train.py: Contains the training loop, validating loop, and testing loop
- predict.py: Gives an example on the usage of a trained ViT

Commit:
Initial Commit

Commit:
Started vit, loaded dataset

Commit:
Added gitignore

Commit:
Patch embedding working for singular image

Commit:
Flattened patches

Commit:
Reshaped to get dimensions from paper

Commit:
Implemented patch embedding torch module

Commit:
Fixed minor bug

Commit:
Prepended random tensor to embedding

Commit:
z_0 (Eq. 1) fully finished

Commit:
Started on ViT module

Commit:
Added transformer encoder to ViT module

Commit:
Implemented Transformer encoder as its own module

Commit:
Implemented the MLP head

Commit:
Combined modules to implement ViT

Commit:
Training code finished

Commit:
Testing code finished, fixed batch size bug

Commit:
different vit versions, added gitignore

Commit:
Added cleaned vit file;Last commit before overhaul

Commit:
added dataset, modules, predict, train

Commit:
50 -> 57% accuracy

Commit:
Hyperparameters tuning, 57% with lower epochs

Commit:
Transfered project to fork

Commit:
Removed weight decay -> 61% accuracy

Commit:
Modularised Files Finished v1

Commit:
embed dim updated to 256 (grayscale)

Commit:
Reverted to 768 embed dimension

Commit:
Restored patient level split

STommy64 committed:
Update README.md

Added ViT description, problem description, how ViT works, dependencies, data preprocessing information.

Commit:
Final predict.py, train.py, dataset.py

STommy64 committed:
Update README.md

Added results, older commits

STommy64 committed:
Delete vit.py

STommy64 committed:
Delete vit_cleaned.py

LinfengLiu98 commented:
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: 
- Goos
- Only the val accuracy plot is shown. No training accuracy and validation accuracy.
- Need to compare training acc with val acc to tell how bad the overfitting is


Commit messages: good

Code:
- Uses Vision transformer
- Well commented
- Good design

Functionality/Performance:
- Did patient-level split
- Model is overfitting
- Test acc is 61.62%


General comments:
**PR is in the main branch**

- Solves the problem appropriately although performance is poor

STommy64 committed:
Final readme

shakes76 commented:
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present, need to use own folder for report code, restore repo README -3
No Usage & Demo & Visualisation & Data usage -1
Module present
Commenting minimal -1
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, no training plot -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, need to move code into own folder, restore repo README  -2
Request Description OK

shakes76 commented:
Feedback marks possible +2 if the requested changes are made (see above). Also remove gitignore to allow merge

STommy64 committed:
Delete .gitignore

Commit:
Merge branch 'topic-recognition' of github.com:STommy64/PatternAnalysis-2023 into topic-recognition

Commit:
Delete redundant files

Commit:
Added ViT folder

Commit:
Delete redundant files

Commit:
Added license into report folder

Commit:
Merge branch 'topic-recognition' of github.com:STommy64/PatternAnalysis-2023 into topic-recognition

Commit:
Moved original repo readme to root

STommy64 commented:
[image: image.png]
Moved original readme into root and moved existing code and report readme
into new folder
[image: image.png]

On Sat, Nov 18, 2023 at 11:10 AM Shakes ***@***.***> wrote:

> Feedback marks possible +2 if the requested changes are made (see above).
> Also remove gitignore to allow merge
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/shakes76/PatternAnalysis-2023/pull/68#issuecomment-1817304015>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BBX4HXLAAZVVH4V6AXWBJM3YFADJBAVCNFSM6AAAAAA6ICXSS2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMJXGMYDIMBRGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

wangzhaomxy commented:
Feedback attempt and feedback marks granted +2, though the adjustment is not fully successful. Should move the student-created-document into topic-recognition/recognition/, instead of just topic-recognition/. But give granted marks for your attempt. Try to fix it again so that Shakes can merge your pull request.

Commit:
renamed to recognition

Commit:
Added nested folder

STommy64 committed:
Delete README.md

STommy64 commented:
Hi, is the following correct?
[image: image.png]

On Tue, Nov 21, 2023 at 3:17 PM Zhao WANG ***@***.***> wrote:

> Feedback attempt and feedback marks granted +2, though the adjustment is
> not fully successful. Should move the student-created-document into
> topic-recognition/recognition/, instead of just topic-recognition/. But
> give granted marks for your attempt. Try to fix it again so that Shakes can
> merge your pull request.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/shakes76/PatternAnalysis-2023/pull/68#issuecomment-1820256346>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BBX4HXPKWD6J7RTLIPEW2HTYFQ2N5AVCNFSM6AAAAAA6ICXSS2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMRQGI2TMMZUGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
